packages/peregrine/lib/talons/Link/useLink.js

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useRef } from 'react';
+import { useLazyQuery } from '@apollo/client';
+import useIntersectionObserver from '../../hooks/useIntersectionObserver';
+import resourceUrl from '../../util/makeUrl';
+import mergeOperations from '../../util/shallowMerge';
+import DEFAULT_OPERATIONS from '../MagentoRoute/magentoRoute.gql';
+
+const useLink = (props, passedOperations = {}) => {
+    const { prefetchType: shouldPrefetch, innerRef: originalRef, to } = props;
+    const operations = shouldPrefetch
+        ? mergeOperations(DEFAULT_OPERATIONS, passedOperations)
+        : {};
+
+    const intersectionObserver = useIntersectionObserver();
+    const { resolveUrlQuery } = operations;
+    const generatedRef = useRef();
+    const elementRef =
+        originalRef || !shouldPrefetch ? originalRef : generatedRef;
+    const [runQuery, { called: pageTypeCalled }] = useLazyQuery(
+        resolveUrlQuery
+    );
+    const linkPath = shouldPrefetch ? resourceUrl(to) : null;
+
+    useEffect(() => {
+        if (
+            !shouldPrefetch ||
+            pageTypeCalled ||
+            !runQuery ||
+            !elementRef.current ||
+            typeof intersectionObserver === 'undefined'
```

Comment:
Since you already perform this check in `useIntersectionObserver`, you can probably simplify.

```suggestion
            !intersectionObserver
```